gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ResourceManager.java

```diff
@@ -27,5 +33,214 @@
 
   public static final String DEFAULT_CONTENT_TYPE = "application/octet-stream";
 
-  // TODO(ajaykannan): Fix me! Add in missing methods.
+  /**
+   * The fields of a project.
+   *
+   * These values can be used to specify the fields to include of in a partial response when calling
+   * {@link ResourceManager#get} or {@link ResourceManager#list}. Project ID is always returned,
+   * even if not specified.
+   */
+  enum ProjectField {
+    PROJECT_ID("projectId"),
+    NAME("name"),
+    LABELS("labels"),
+    PROJECT_NUMBER("projectNumber"),
+    STATE("lifecycleState"),
+    CREATE_TIME("createTime");
+
+    private final String selector;
+
+    ProjectField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+
+    static String selector(ProjectField... fields) {
+      HashSet<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(PROJECT_ID.selector());
+      for (ProjectField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * Class for specifying project get options.
+   */
+  public class ProjectGetOption extends Option {
+
+    private static final long serialVersionUID = 270185129961146874L;
+
+    private ProjectGetOption(ResourceManagerRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the project's fields to be returned by the RPC call.
+     *
+     * If this option is not provided all project fields are returned.
+     * {@code ProjectGetOption.fields} can be used to specify only the fields of interest. Project
+     * ID is always returned, even if not specified. {@link ProjectField} provides a list of fields
+     * that can be used.
+     */
+    public static ProjectGetOption fields(ProjectField... fields) {
+      return new ProjectGetOption(ResourceManagerRpc.Option.FIELDS, ProjectField.selector(fields));
+    }
+  }
+
+  /**
+   * Class for specifying project list options.
+   */
+  public class ProjectListOption extends Option {
```

Comment:
We decided to leave them out, and I created issue #421 to add them in when ready.